rpc/core/pipe.go

```diff
@@ -89,6 +90,10 @@ func SetTxIndexer(indexer txindex.TxIndexer) {
 	txIndexer = indexer
 }
 
+func SetConsensusReactor(conR *consensus.ConsensusReactor) {
+	consensusReactor = conR
```

Comment:
Same thing here that @ebuchman had raised about racy behavior, perhaps we need a global mutex of sorts since this function mutates a global consensusReactor.